src/Microsoft.AspNetCore.DataProtection.EntityFrameworkCore/EntityFrameworkCoreXmlRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.AspNetCore.DataProtection.Repositories;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNetCore.DataProtection.EntityFrameworkCore
+{
+    /// <summary>
+    /// An <see cref="IXmlRepository"/> backed by an EntityFrameworkCore datastore.
+    /// </summary>
+    public class EntityFrameworkCoreXmlRepository<TContext> : IXmlRepository
+        where TContext : DbContext, IDataProtectionKeyContext
+    {
+        private readonly TContext _context;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="EntityFrameworkCoreXmlRepository{TContext}"/>.
+        /// </summary>
+        /// <param name="context">The context used to store instances of <see cref="DataProtectionKey"/></param>
+        public EntityFrameworkCoreXmlRepository(TContext context) => _context = context ?? throw new ArgumentNullException(nameof(context));
```

Comment:
Instead of taking an instance of `TContext`, we should take a service which can construct a new instance of the DbContext. In earlier versions of EFCore, there was an `IDbContextFactory<TContext>` service you could use. I don't remember if that is still around in EF Core 2.0 (cc @divega).

Also, EntityFrameworkCoreXmlRepository is a singleton, but each time it invokes an interaction with the database, we should create a new instance of DbContext.

Reply:
Don't use `IDbContextFactory`. It was only for design-time. We renamed it in 2.0 to reflect that.

Reply:
When are keys retrieved? Once during app start? During a request?

If it's just once, just creating a scope here might be best.